contracts/azorius/BaseVotingBasisPercent.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity =0.8.19;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/**
+ * An Azorius extension contract that enables percent based voting basis calculations.
+ *
+ * Intended to be implemented by BaseStrategy implementations, this allows for voting strategies
+ * to dictate any basis strategy for passing a Proposal between >50% (simple majority) to 100%.
+ *
+ * See https://en.wikipedia.org/wiki/Voting#Voting_basis.
+ * See https://en.wikipedia.org/wiki/Supermajority.
+ */
+abstract contract BaseVotingBasisPercent is OwnableUpgradeable {
+    
+    /** The numerator to use when calculating basis (adjustable). */
+    uint256 public basisNumerator;
+
+    /** The denominator to use when calculating basis (1,000,000). */
+    uint256 public constant BASIS_DENOMINATOR = 1_000_000;
+
+    error InvalidBasisNumerator();
+
+    event BasisNumeratorUpdated(uint256 basisNumerator);
+
+    /**
+     * Calculates the basis percentage for a Proposal on the strategy to pass.
+     *
+     * In a simple majority, the basis would be 50% (500,000 / 1,000,000).
+     *
+     * Adjusting the basisNumerator allows for strategies to use any basis
+     * above simple majority, such as 2/3, 90%, etc.
+     */
+    function basis() public view virtual returns (uint256) {
+        return basisNumerator / BASIS_DENOMINATOR;
+    }
```

Comment:
This math doesn't work in Solidity, since all numbers are integers. This would always just return 0, since the numerator is less than the denominator.

Reply:
I think we could just remove this function, and use the numerator and denominator in our math directly as I showed in the comment below